packages/network-controller/src/rpc-service/rpc-service.ts

```diff
@@ -483,20 +483,48 @@ export class RpcService implements AbstractRpcService {
     return await this.#policy.execute(async () => {
       const response = await this.#fetch(this.endpointUrl, fetchOptions);
 
-      if (response.status === 405) {
+      if (response.status === 401) {
+        // code: -33100
+        const errorData: JsonRpcError = {
+          code: -33100,
+          message: 'Unauthorized.',
+        };
+        const error = new Error('Unauthorized.');
+        Object.assign(error, errorData);
+        throw error;
```

Comment:
If we throw a regular error here, it will end up interpreted as an internal RPC error by `json-rpc-engine` in the end. To be treated as an RPC error, it needs to extend the JsonRpcError class. That class includes a `serialize` method that we need, for example.

If you want to define this new error type inline here, I'd recommend something more like `new JsonRpcError(-33100, 'Unauthorized.')`.

Reply:
Thanks fixed here https://github.com/MetaMask/core/pull/5807/commits/10ff958eed716d6219966f19b7ceaa992d860cf7